fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LucenePartitioner.java

```diff
@@ -295,14 +316,18 @@ byte[] partitionMetadataKeyFromTimestamp(@Nonnull Tuple groupKey, long timestamp
     void savePartitionMetadata(@Nonnull Tuple groupKey, @Nonnull final LucenePartitionInfoProto.LucenePartitionInfo.Builder builder) {
         LucenePartitionInfoProto.LucenePartitionInfo updatedPartition = builder.build();
         state.context.ensureActive().set(
-                partitionMetadataKeyFromTimestamp(groupKey, Tuple.fromBytes(builder.getFrom().toByteArray()).getLong(0)),
+                partitionMetadataKeyFromTimestamp(groupKey, getTimestampFromPartitionInfo(builder)),
                 updatedPartition.toByteArray());
     }
 
+    private static long getTimestampFromPartitionInfo(@Nonnull final LucenePartitionInfoProto.LucenePartitionInfoOrBuilder builder) {
```

Comment:
there is already a static method in this class that does this (`getFrom`, taking a `LucenePartitionInfo` parameter). Maybe these can be combined